src/components/layout/location-context.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { ReactNode, Dispatch, SetStateAction } from 'react'
+import type { ModalPayloadType } from './layout'
+
+type LocationProviderProps = {
+    children: ReactNode
+    has_mounted?: boolean
+    setModalPayload?: Dispatch<SetStateAction<ModalPayloadType>>
+    show_cookie_banner?: boolean
+    toggleModal?: () => void
```

Comment:
```suggestion
    toggleModal?: React.MouseEventHandler<HTMLElement>
```
or something like this, I mean we should provide some kinda type for the event, we may need it in other components. :pray: